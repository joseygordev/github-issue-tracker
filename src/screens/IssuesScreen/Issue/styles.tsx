import Colors from '../../../constants/Colors';
import { Dimensions, StyleSheet } from 'react-native';

export default StyleSheet.create({
  card: {
    backgroundColor: Colors.white,
    paddingHorizontal: 16,
    paddingVertical: 16,
    borderRadius: 8,
    marginBottom: 16,
    elevation: 4,
    shadowColor: Colors.steel,
    shadowOffset: {
      width: 2,
      height: 2,
    },
    shadowRadius: 3,
    shadowOpacity: 0.1,
  },
  profileImage: {
    width: 20,
    height: 20,
    borderRadius: 20 / 2,
    marginRight: 8,
    marginTop: 1,
  },
  user: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 16,
    marginTop: -2,
  },
  username: {
    color: Colors.steel,
    fontSize: 12,
  },
  title: {
    maxWidth: Dimensions.get('screen').width - 16,
  },
  labels: {
    flexDirection: 'row',
    marginTop: 8,
    flexWrap: 'wrap',
  },
  label: {
    borderRadius: 16,
    paddingHorizontal: 10,
    height: 22,
    alignItems: 'center',
    flexDirection: 'row',
    marginRight: 8,
    marginTop: 8,
  },
  labelText: {
    fontSize: 12,
    color: Colors.steel
  },
  timeAgo: {
    position: 'absolute',
    top: 10,
    right: 14,
  },
  timeAgoText: {
    color: Colors.border,
    fontSize: 12,
  },
  state: {
    position: 'absolute',
    right: 0,
    top: 13,
    paddingRight: 14,
    paddingLeft: 12,
    paddingVertical: 4,
    borderTopLeftRadius: 16,
    borderBottomLeftRadius: 16,
  },
  stateText: {
    color: Colors.white,
  },
  star: {
  },
});
