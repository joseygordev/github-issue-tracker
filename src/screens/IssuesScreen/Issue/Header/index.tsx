import React from 'react';
import moment from 'moment';
import { View, Image } from 'react-native';
import { AntDesign } from '@expo/vector-icons';

import { IssueHeaderProps } from '../../../../types/issues';
import Colors from '../../../../constants/Colors';

import CustomText from '../../../../components/CustomText';

import styles from './styles';

export default function Header(props: IssueHeaderProps) {
  const { issue } = props;

  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <Image style={styles.profileImage} source={{ uri: issue.user.avatar_url }} />
        <CustomText style={styles.username}>{issue.user.login}</CustomText>
        <CustomText style={styles.timeAgoText}> • {moment.duration(moment().diff(moment(issue.created_at))).humanize() + ' ago'}</CustomText>
        {issue.isBookmarked && (
          <>
            <CustomText style={styles.timeAgoText}> • </CustomText>
            <AntDesign size={16} color={Colors.border} name="star" />
          </>
        )}
      </View>
      <View style={[styles.state, { backgroundColor: issue.state == 'open' ? Colors.green : Colors.border }]}>
        <CustomText style={styles.stateText}>{issue.state[0].toUpperCase() + issue.state.slice(1)}</CustomText>
      </View>
    </View>
  );
}
