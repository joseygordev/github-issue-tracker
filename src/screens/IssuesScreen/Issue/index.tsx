import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';

import { IssueItemProps } from '../../../types/issues';

import CustomText from '../../../components/CustomText';
import Header from './Header';

import styles from './styles';

export default function Issue(props: IssueItemProps) {
  const navigation = useNavigation();

  const { issue } = props;

  const goToDetails = () => {
    navigation.navigate('IssueDetails', {
      issue: issue
    })
  };

  return (
    <TouchableOpacity testID="cardButton" activeOpacity={0.8} onPress={goToDetails} style={styles.card}>
      <Header issue={issue} />
      <CustomText style={styles.title}>{issue.title}</CustomText>
      <View style={styles.labels}>
        {issue.labels.map((label: any) => (
          <View key={label.id} style={[styles.label, { backgroundColor: '#' + label.color }]}>
            <CustomText style={styles.labelText}>
              {label.name.replace(/\:([a-z]+)\:/g, '')}
            </CustomText>
          </View>
        ))}
      </View>
    </TouchableOpacity>
  );
}
