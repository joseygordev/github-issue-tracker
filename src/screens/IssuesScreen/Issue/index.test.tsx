import React from 'react';
import { create } from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';

import mockIssues from '../../../data/mocks/issues';
import Issue from './index';

describe('Issue', () => {
  const issue = mockIssues[0];
  const createFreshTree = () =>
      create(
        <NavigationContainer>
          <Issue navigation={{}} issue={issue} />
        </NavigationContainer>
      );
  const treeComponent = createFreshTree();

  it('it renders Issue correctly', async () => {
    expect(treeComponent).toMatchSnapshot();
  });
  
  it('it renders Issue with correctly title', async () => {
    const issueComponent = treeComponent.root.findByProps({ testID: 'issueTitle' }).props;
  
    expect(issueComponent.children).toEqual(issue.title);
  });
});
