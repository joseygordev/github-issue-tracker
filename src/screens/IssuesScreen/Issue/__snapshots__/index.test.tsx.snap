// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue it renders Issue correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 8,
      "elevation": 4,
      "marginBottom": 16,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "shadowColor": "#414d5b",
      "shadowOffset": Object {
        "height": 2,
        "width": 2,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 3,
    }
  }
  testID="cardButton"
>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "mock_url",
          }
        }
        style={
          Object {
            "borderRadius": 10,
            "height": 20,
            "marginRight": 8,
            "marginTop": 1,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "color": "#414d5b",
              "fontSize": 12,
            },
            Object {
              "fontFamily": "space-mono",
            },
          ]
        }
        testID="userName"
      >
        mock_login
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "color": "#aab2ba",
              "fontSize": 12,
            },
            Object {
              "fontFamily": "space-mono",
            },
          ]
        }
        testID="textValue"
      >
         â€¢ 
        a few seconds ago
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 16,
            "marginVertical": 8,
            "maxWidth": 80,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingVertical": 4,
          },
          Object {
            "backgroundColor": "#16c65d",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "color": "#fff",
              "textAlign": "center",
            },
            Object {
              "fontFamily": "space-mono",
            },
          ]
        }
        testID="textValue"
      >
        Open
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "maxWidth": 734,
        },
        Object {
          "fontFamily": "space-mono",
        },
      ]
    }
    testID="issueTitle"
  >
    Mock
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "flexDirection": "row",
            "height": 22,
            "marginRight": 8,
            "marginTop": 8,
            "paddingHorizontal": 10,
          },
          Object {
            "backgroundColor": "##000",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "color": "#414d5b",
              "fontSize": 12,
            },
            Object {
              "fontFamily": "space-mono",
            },
          ]
        }
        testID="textValue"
      >
        mock label
      </Text>
    </View>
  </View>
</View>
`;
